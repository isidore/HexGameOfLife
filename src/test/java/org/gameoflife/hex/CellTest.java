package org.gameoflife.hex;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class CellTest {
    @Test
    void oddsGoTogether() {
        assertEquals(true, Cell.isValid(1,1));
        assertEquals(true, Cell.isValid(2,2));
        assertEquals(false, Cell.isValid(2,1));
        assertEquals(false, Cell.isValid(1,2));
    }

    @Test
    void getLevelOneNeighbours() {
        GameOfLifeBoard gameOfLifeBoard = BoardTest.createBoardWithNeighbours(6, 0);
        Approvals.verify(HexPrinter.print(gameOfLifeBoard));
    }
    @Test
    void getLevelTwoNeighbours() {
        GameOfLifeBoard gameOfLifeBoard = BoardTest.createBoardWithNeighbours(0, 6);
        Approvals.verify(HexPrinter.print(gameOfLifeBoard));
    }

}
