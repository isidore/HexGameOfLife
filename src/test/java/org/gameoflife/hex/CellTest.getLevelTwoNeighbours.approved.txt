| = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = =  |
|    _ _           _ _           _ _           _ _           _ _           |
|  /     \       /     \       /     \       /     \       /     \         |
| /  0,0  \ _ _ /  2,0  \ _ _ /  4,0  \ _ _ /  6,0  \ _ _ /  8,0  \ _ _    |
| \       /     \       /X X X\       /X X X\       /     \       /     \  |
|  \ _ _ /  1,1  \ _ _ /X  X  X\ _ _ /X  X  X\ _ _ /  7,1  \ _ _ /  9,1  \ |
|  /     \       /     \X  X  X/     \X  X  X/     \       /     \       / |
| /  0,2  \ _ _ /  2,2  \X_X_X/  4,2  \X_X_X/  6,2  \ _ _ /  8,2  \ _ _ /  |
| \       /     \       /     \       /     \       /     \       /     \  |
|  \ _ _ /  1,3  \ _ _ /  3,3  \ _ _ /  5,3  \ _ _ /  7,3  \ _ _ /  9,3  \ |
|  /     \       /X X X\       /X X X\       /X X X\       /     \       / |
| /  0,4  \ _ _ /X  X  X\ _ _ /X  X  X\ _ _ /X  X  X\ _ _ /  8,4  \ _ _ /  |
| \       /     \X  X  X/     \X  X  X/     \X  X  X/     \       /     \  |
|  \ _ _ /  1,5  \X_X_X/  3,5  \X_X_X/  5,5  \X_X_X/  7,5  \ _ _ /  9,5  \ |
|  /     \       /     \       /     \       /     \       /     \       / |
| /  0,6  \ _ _ /  2,6  \ _ _ /  4,6  \ _ _ /  6,6  \ _ _ /  8,6  \ _ _ /  |
| \       /     \       /X X X\       /X X X\       /     \       /     \  |
|  \ _ _ /  1,7  \ _ _ /X  X  X\ _ _ /X  X  X\ _ _ /  7,7  \ _ _ /  9,7  \ |
|  /     \       /     \X  X  X/     \X  X  X/     \       /     \       / |
| /  0,8  \ _ _ /  2,8  \X_X_X/  4,8  \X_X_X/  6,8  \ _ _ /  8,8  \ _ _ /  |
| \       /     \       /     \       /     \       /     \       /     \  |
|  \ _ _ /  1,9  \ _ _ /  3,9  \ _ _ /  5,9  \ _ _ /  7,9  \ _ _ /  9,9  \ |
|        \       /     \       /     \       /     \       /     \       / |
|         \ _ _ /       \ _ _ /       \ _ _ /       \ _ _ /       \ _ _ /  |
|                                                                          |
| = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = =  |
